### PR TITLE
A11y/Dashboard: Change settings nav items from buttons to links

### PR DESCRIPTION
**Pull request by kaydelaney**
**What this PR does / why we need it**:
Changes dashboard settings navigation items so that rather than using anchor tags with `onClick` events (which prevents tab-navigation and URL previews), we use react-router links which render as anchor tags but provide the missing features mentioned previously.

### CODE REVIEW

File: public/app/features/dashboard/components/DashboardSettings/DashboardSettings.tsx [hunk 1]
```diff
@@ -26,44 +26,67 @@ export interface SettingsPage {
...
 
-export class DashboardSettings extends PureComponent<Props> {
-  onClose = () => {
-    locationService.partial({ editview: null });
-  };
-
-  onChangePage = (editview: string) => {
-    locationService.partial({ editview });
-  };
+const updateSearchParams = (init: string, partial: string) => {
```
Comment (Clarity-89) [inline public/app/features/dashboard/components/DashboardSettings/DashboardSettings.tsx:32]: I think we have a utility function that does smth similar: https://github.com/grafana/grafana/blob/main/packages/grafana-data/src/utils/location.ts#L39

File: public/app/features/dashboard/components/DashboardSettings/DashboardSettings.tsx [hunk 2]
```diff
@@ -26,44 +26,67 @@ export interface SettingsPage {
...
+  const newSearchParams = new URLSearchParams(partial);
+  newSearchParams.forEach((v, k) => {
+    searchParams.set(k, v);
+  });
+  return '?' + searchParams.toString();
+};
+
+const onClose = () => locationService.partial({ editview: null });
+
+const MakeEditable: React.FC<{ onMakeEditable: () => any }> = (props) => (
```
Comment (Clarity-89) [inline public/app/features/dashboard/components/DashboardSettings/DashboardSettings.tsx:43]: I thought we were moving away from using `React.FC? `

Comment (kaydelaney) [reply public/app/features/dashboard/components/DashboardSettings/DashboardSettings.tsx:43]: Ah right, will change.

**Review by Clarity-89 — APPROVED** [review]

**Review by axelavargas — APPROVED** [review]
Nice work @kaydelaney , I also gave it a spin, and works as expected 🥳

**Pull request merged**